src/components/ScreenWrapper/index.js

```diff
@@ -1,148 +1,22 @@
-        }
-        if (this.unsubscribeTransitionEnd) {
-            this.unsubscribeTransitionEnd();
-        }
-        if (this.beforeRemoveSubscription) {
-            this.beforeRemoveSubscription();
-        }
-    }
-
-    render() {
-        return (
-            <SafeAreaConsumer>
-                {({
-                    insets, paddingTop, paddingBottom, safeAreaPaddingBottomStyle,
-                }) => {
-                    const paddingStyle = {};
-
-                    if (this.props.includePaddingTop) {
-                        paddingStyle.paddingTop = paddingTop;
-                    }
-
-                    // We always need the safe area padding bottom if we're showing the offline indicator since it is bottom-docked.
-                    if (this.props.includeSafeAreaPaddingBottom || this.props.network.isOffline) {
-                        paddingStyle.paddingBottom = paddingBottom;
-                    }
+import BaseScreenWrapper from './BaseScreenWrapper';
+import withEnvironment from '../withEnvironment';
+import withWindowDimensions from '../withWindowDimensions';
 
-                    return (
-                        <View
-                            style={[
-                                ...this.props.style,
-                                styles.flex1,
-                                paddingStyle,
-                            ]}
-                        >
-                            <KeyboardAvoidingView style={[styles.w100, styles.h100, {maxHeight: this.props.windowHeight}]} behavior={this.props.keyboardAvoidingViewBehavior}>
-                                <HeaderGap />
-                                {// If props.children is a function, call it to provide the insets to the children.
-                                    _.isFunction(this.props.children)
-                                        ? this.props.children({
-                                            insets,
-                                            safeAreaPaddingBottomStyle,
-                                            didScreenTransitionEnd: this.state.didScreenTransitionEnd,
-                                        })
-                                        : this.props.children
-                                }
-                                {this.props.isSmallScreenWidth && (
-                                    <OfflineIndicator />
-                                )}
-                            </KeyboardAvoidingView>
-                        </View>
-                    );
-                }}
-            </SafeAreaConsumer>
-        );
-    }
-}
+const ScreenWrapper = props => (
+    <BaseScreenWrapper
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+    />
+);
 
 ScreenWrapper.propTypes = propTypes;
 ScreenWrapper.defaultProps = defaultProps;
+ScreenWrapper.displayName = 'ScreenWrapper';
 
 export default compose(
-    withNavigation,
+    withEnvironment,
```

Comment:
We need this `withEnvironment` because?

Reply:
It's passed through to BaseScreenWrapper and used to determine whether we are using dev environment

Reply:
got it, then why not move it to `BaseScreenWrapper`? why pass it twice for index.js and index.native.js ?